src/databricks/labs/lsql/dashboards.py

```diff
@@ -146,9 +186,7 @@ def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
     def _get_datasets(dashboard_folder: Path) -> list[Dataset]:
         datasets = []
         for query_path in sorted(dashboard_folder.glob("*.sql")):
-            with query_path.open("r") as query_file:
-                raw_query = query_file.read()
-            dataset = Dataset(name=query_path.stem, display_name=query_path.stem, query=raw_query)
+            dataset = Dataset(name=query_path.stem, display_name=query_path.stem, query=query_path.read_text())
```

Comment:
technically, you should 

```
class Tile:
    def __init__(self, path: Path):
```

and do parsing/dataset creation there, so that query-specific parsing is handled in the `Tile` subclasses and `Dashboards` class has more high-level logic for combining it all. this way it's a bit more extendable. This is the direction i want codebase to move into.

Reply:
See my changes